src/js/client/editprefset-settingsTable-widget.js

```diff
@@ -0,0 +1,259 @@
+    bindings: { // Binding selectors to model paths
+    },
+    selectors: {
+        valueDisplayCell: ".pmt-value-display",
+        enabledBooleanInputs: ".pmt-enabled-boolean",
+        settingsFilter: "#filter-container",
+        settingsRows: ".pmt-settings-table-row",
+        addContextButton: ".pmt-add-context-button",
+        mineAllSwitchContainer: "#mineAllSwitch-container"
+    },
+    templates: {
+        initial: "editprefset-settingsTable-widget"
+    },
+    components: {
+        mineAllSwitch: {
+            type: "gpii.devpmt.mineAllSwitch.default",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.mineAllSwitchContainer",
+            options: {
+                selectors: {
+                    initial: "#mineAllSwitch-area"
+                },
+                model: {
+                    settingsFilter: "{settingsTableWidget}.model.allSettingsEnabled",
+                    allSettingsCount: "{settingsTableWidget}.model.termUsage.all",
+                    mySettingsCount: "{settingsTableWidget}.model.termUsage.npset"
+                }
+            }
+        },
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.settingsFilter",
+            options: {
+                selectors: {
+                    initial: "#common-filter-area"
+                },
+                model: {
+                    filterText: "{settingsTableWidget}.model.settingsFilter"
+                }
+            }
+        }
+    },
+    invokers: {
+        filterInit: {
+            funcName: "gpii.devpmt.settingsTable.filterInit",
+            args: ["{that}"]
+        },
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}"]
+        },
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.settingsTable.updateLunrIndex",
+            args: ["{that}"]
+        },
+        filterSettings: {
+            funcName: "gpii.devpmt.settingsTable.filterSettings",
+            args: ["{that}", "{that}.dom.settingsRows", "{that}.lunrIndex", "{that}.model.settingsFilter"]
+        },
+        enableProductListener: {
+            funcName: "gpii.devpmt.settingsTable.enableProductListener",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"] // event
+        },
+        updateTermUsage: {
+            funcName: "gpii.devpmt.settingsTable.updateTermUsage",
+            args: ["{that}"]
+        }
+    },
+    listeners: {
+        "onCreate": [
+            {
+                "func": "{that}.updateLunrIndex"
+            },
+            {
+                "func": "{that}.filterInit"
+            },
+            {
+                "func": "{that}.updateTermUsage"
+            }
+        ],
+        "onMarkupRendered": [
+            {
+                "this": "{that}.dom.enabledBooleanInputs",
```

Comment:
Probably better with delegated binding, plus a conversion from DOM element to data path, rather than embedding the whole "dataset" in the DOM